### PR TITLE
chore(ast): support expr in the position of function parameter

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._

Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14122)

### CODE REVIEW

File: src/query/functions/src/scalars/decimal/math.rs [hunk 1]
```diff
@@ -261,13 +261,13 @@ fn decimal_math(
         return arg.clone().to_owned();
     }
 
-    let none_negative = target_scale >= 0;
+    let positive = target_scale >= 0;
```
Comment (sundy-li) [inline src/query/functions/src/scalars/decimal/math.rs:264]: is zero positive?

File: src/query/functions/tests/it/aggregates/testdata/agg.txt [hunk 2]
```diff
@@ -823,18 +823,18 @@ evaluation (internal):
...
-+--------+-----------------------------------------------------------------+
-| a      | Int64([4, 3, 2, 1])                                             |
-| Output | NullableColumn { column: Float64([1]), validity: [0b_______1] } |
-+--------+-----------------------------------------------------------------+
++--------+-------------------------------------------------------------------+
+| Column | Data                                                              |
++--------+-------------------------------------------------------------------+
+| a      | Int64([4, 3, 2, 1])                                               |
+| Output | NullableColumn { column: Float64([3.4]), validity: [0b_______1] } |
++--------+-------------------------------------------------------------------+
```
Comment (sundy-li) [inline src/query/functions/tests/it/aggregates/testdata/agg.txt:837]: cc @ariesdevil  so the `quantile_cont` is wrong without this pr ?

**Review by sundy-li — APPROVED** [review]
Need fix UT. Rest looks good to me

File: src/query/functions/tests/it/aggregates/testdata/agg.txt (hunk 2 above)
Comment (ariesdevil) [reply src/query/functions/tests/it/aggregates/testdata/agg.txt:837]: `quantile_cont` compute results are correct in actual execution, but it is wrong in agg.txt.

**Pull request merged**